autofit/optimize/non_linear/samples.py

```diff
@@ -21,39 +22,111 @@ def __init__(self, model, paths):
-        raise NotImplementedError()
+    def max_log_likelihood_index(self) -> int:
+        """The index of the sample with the highest log likelihood."""
+        return int(np.argmax(self.log_likelihoods))
 
     @property
-    def most_likely_vector(self) -> [float]:
-        """ The best-fit model sampled by the non-linear search (corresponding to the maximum log-likelihood), returned
-        as a list of values."""
-        raise NotImplementedError()
+    def max_log_likelihood_vector(self) -> [float]:
+        """ The parameters of the maximum log likelihood sample of the non-linear search returned as a list of values."""
+        return self.parameters[self.max_log_likelihood_index]
 
     @property
-    def most_likely_instance(self) -> model.ModelInstance:
-        """ The best-fit model sampled by the non-linear search (corresponding to the maximum log-likelihood), returned
-        as a model instance."""
-        return self.model.instance_from_vector(vector=self.most_likely_vector)
+    def max_log_likelihood_instance(self) -> model.ModelInstance:
+        """  The parameters of the maximum log likelihood sample of the non-linear search returned as a model instance."""
+        return self.model.instance_from_vector(vector=self.max_log_likelihood_vector)
+
+    @property
+    def max_log_posterior_index(self) -> int:
+        """The index of the sample with the highest log posterior."""
+        return int(np.argmax(self.log_posteriors))
+
+    @property
+    def max_log_posterior_vector(self) -> [float]:
+        """ The parameters of the maximum log posterior sample of the non-linear search returned as a list of values."""
+        return self.parameters[self.max_log_posterior_index]
+
+    @property
+    def max_log_posterior_instance(self) -> model.ModelInstance:
+        """  The parameters of the maximum log posterior sample of the non-linear search returned as a model instance."""
+        return self.model.instance_from_vector(vector=self.max_log_posterior_vector)
+
+    @property
+    def pdf_converged(self) -> bool:
+        """ To analyse and visualize chains using *GetDist*, the analysis must be sufficiently converged to produce
+        smooth enough PDF for analysis. This property checks whether the non-linear search's chains are sufficiently
+        converged for *GetDist* use.
+
+        For *Dynesty*, during initial sampling one accepted live point typically has > 99% of the probabilty as its
+        log_likelihood is significantly higher than all other points. Convergence is only achieved late in sampling when
+        all live points have similar log_likelihood and sampling probabilities."""
+        try:
+            densities_1d = list(
+                map(lambda p: self.pdf.get1DDensity(p), self.pdf.getParamNames().names)
+            )
+
+            if densities_1d == []:
+                return False
```

Comment:
This looks a lot like another method I've seen recently. I'm guessing this is repetition between samplers we could maybe avoid?

Reply:
Will sort, its all annoying GetDist stuff so requires some thought.